### PR TITLE
[FLASK] Create E2E test for snap_dialog

### LINKED ISSUE #17107: [FLASK] Create E2E Test for snap_dialog
This will create an E2E test for snap_dialog

### PR DESCRIPTION
**Pull request by bowensanders**
This creates a new E2E test for snap_dialog.
It tests all three current functions of snap_dialog - alert, confirmation and prompt.

Fixes #17107

### CODE REVIEW

File: test/e2e/snaps/test-snap-dialog.spec.js [hunk 1]
```diff
@@ -0,0 +1,201 @@
...
+        windowHandles = await driver.waitUntilXWindowHandles(2, 1000, 10000);
+        await driver.switchToWindowWithTitle('Test Snaps', windowHandles);
+
+        // click on alert dialog
+        await driver.clickElement('#sendAlertButton');
+        await driver.delay(1000);
+
+        // switch to dialog popup
+        windowHandles = await driver.waitUntilXWindowHandles(3, 1000, 10000);
+        windowHandles = await driver.getAllWindowHandles();
```
Comment (FrederikBolding) [inline test/e2e/snaps/test-snap-dialog.spec.js:84]:
```suggestion
```

File: test/e2e/snaps/test-snap-dialog.spec.js [hunk 2]
```diff
@@ -0,0 +1,201 @@
...
+        await driver.delay(1000);
+        assert.equal(await result.getText(), 'null');
+
+        // click conf button
+        await driver.clickElement('#sendConfButton');
+        await driver.delay(1000);
+
+        // switch to dialog popup
+        windowHandles = await driver.waitUntilXWindowHandles(3, 1000, 10000);
+        windowHandles = await driver.getAllWindowHandles();
```
Comment (FrederikBolding) [inline test/e2e/snaps/test-snap-dialog.spec.js:118]:
```suggestion
```

File: test/e2e/snaps/test-snap-dialog.spec.js [hunk 3]
```diff
@@ -0,0 +1,201 @@
...
+        await driver.delay(1000);
+        assert.equal(await result.getText(), 'false');
+
+        // click conf button again
+        await driver.clickElement('#sendConfButton');
+        await driver.delay(1000);
+
+        // switch to dialog popup
+        windowHandles = await driver.waitUntilXWindowHandles(3, 1000, 10000);
+        windowHandles = await driver.getAllWindowHandles();
```
Comment (FrederikBolding) [inline test/e2e/snaps/test-snap-dialog.spec.js:146]:
```suggestion
```

File: test/e2e/snaps/test-snap-dialog.spec.js [hunk 4]
```diff
@@ -0,0 +1,201 @@
...
+        await driver.delay(1000);
+        assert.equal(await result.getText(), 'true');
+
+        // click prompt button
+        await driver.clickElement('#sendPromptButton');
+        await driver.delay(1000);
+
+        // switch to dialog popup
+        windowHandles = await driver.waitUntilXWindowHandles(3, 1000, 10000);
+        windowHandles = await driver.getAllWindowHandles();
```
Comment (FrederikBolding) [inline test/e2e/snaps/test-snap-dialog.spec.js:174]:
```suggestion
```

File: test/e2e/snaps/test-snap-dialog.spec.js [hunk 5]
```diff
@@ -0,0 +1,201 @@
...
+          'MetaMask Notification',
+          windowHandles,
+        );
+        await driver.delay(1000);
+
+        // fill '2323' in form field
+        await driver.fill('.MuiInput-input', '2323');
+
+        // click submit button
+        await driver.clickElement({
```
Comment (FrederikBolding) [inline test/e2e/snaps/test-snap-dialog.spec.js:185]: Should we test the reject case for the prompt as well?

File: test/e2e/snaps/test-snap-dialog.spec.js (hunk 3 above)
Comment (bowensanders) [reply test/e2e/snaps/test-snap-dialog.spec.js]: fixed in latest commit

File: test/e2e/snaps/test-snap-dialog.spec.js (hunk 4 above)
Comment (bowensanders) [reply test/e2e/snaps/test-snap-dialog.spec.js]: fixed in latest commit

File: test/e2e/snaps/test-snap-dialog.spec.js (hunk 5 above)
Comment (bowensanders) [reply test/e2e/snaps/test-snap-dialog.spec.js:213]: I've added this in the latest commit

Comment (kingdutch001) [reply test/e2e/snaps/test-snap-dialog.spec.js:213]: Test

File: test/e2e/snaps/test-snap-dialog.spec.js (hunk 3 above)
Comment (kingdutch001) [reply test/e2e/snaps/test-snap-dialog.spec.js]: Good job

**Review by FrederikBolding — DISMISSED** [review]

File: test/e2e/snaps/test-snap-dialog.spec.js [hunk 6]
```diff
@@ -0,0 +1,229 @@
...
+        windowHandles = await driver.waitUntilXWindowHandles(2, 1000, 10000);
+        await driver.switchToWindowWithTitle('Test Snaps', windowHandles);
+
+        // click on alert dialog
+        await driver.clickElement('#sendAlertButton');
+        await driver.delay(1000);
+
+        // switch to dialog popup
+        windowHandles = await driver.waitUntilXWindowHandles(3, 1000, 10000);
+        // windowHandles = await driver.getAllWindowHandles();
```
Comment (GuillaumeRx) [inline test/e2e/snaps/test-snap-dialog.spec.js:84]: Do we still need this comment ?

Comment (GuillaumeRx) [reply test/e2e/snaps/test-snap-dialog.spec.js:84]: There is multiple occurence of this comment in the file

**Review by GuillaumeRx — APPROVED** [review]

**Review by FrederikBolding — APPROVED** [review]

**Pull request merged**